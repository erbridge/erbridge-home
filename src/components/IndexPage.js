import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

class IndexPage extends Component {
  static propTypes = {
    routes: PropTypes.arrayOf(
      PropTypes.shape({
        exact: PropTypes.bool,
        path: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
      }),
    ).isRequired,
  };

  render() {
    const { routes } = this.props;

    return (
      <div className="IndexPage">
        {routes.map((route, index) => (
          <div key={route.name || index}>
            <NavLink to={route.path} exact={route.exact}>
              {route.title}
            </NavLink>
          </div>
        ))}
      </div>
    );
  }
}

export default IndexPage;
