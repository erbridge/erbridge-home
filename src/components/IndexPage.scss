@import 'styles/colours';
@import 'styles/layout';
@import 'styles/transition-timings.scss';

$body-padding: 18px;

$link-width: 234px;
$link-height: 270px;
$link-default-margin: 10px;
$link-container-width: $link-width + 2 * $link-default-margin;
$link-short-row-vertical-margin: -1 * $link-width / 4;
$link-short-row-horizontal-margin: $link-width / 2 + 2 * $link-default-margin;

$three-column-min-device-width: 3 * $link-container-width + 2 * $body-padding;
$two-column-min-device-width: 2 * $link-container-width + 2 * $body-padding;
$one-half-column-min-device-width: 1.5 * $link-container-width + 2 *
  $body-padding;

.IndexPage {
  display: flex;
  padding: 8px 0;
  justify-content: center;
}

.IndexPage__content {
  display: flex;
  width: 100%;
  flex-wrap: wrap;

  @include media-min-width($one-half-column-min-device-width) {
    max-width: 1.5 * $link-container-width;
  }

  @include media-min-width($two-column-min-device-width) {
    max-width: 2 * $link-container-width;
  }

  @include media-min-width($three-column-min-device-width) {
    max-width: 3 * $link-container-width;
  }
}

.IndexPage__link {
  position: relative;
  width: 100%;
  height: 120px;
  margin: $link-default-margin;
  padding: 0 8px;
  background-color: $darkest-grey;

  &:active,
  &:focus,
  &:hover {
    .IndexPage__link__image {
      opacity: 0.4;
    }
  }

  $clip-path: polygon(50% 0%, 100% 25%, 100% 75%, 50% 100%, 0% 75%, 0% 25%);

  @supports (clip-path: $clip-path) {
    @include media-min-width($one-half-column-min-device-width) {
      width: $link-width;
      height: $link-height;
      clip-path: $clip-path;

      &:nth-child(2n + 2) {
        margin-top: $link-short-row-vertical-margin;
        margin-left: $link-short-row-horizontal-margin;

        &:not(:last-child) {
          margin-bottom: $link-short-row-vertical-margin;
        }
      }
    }

    @include media-min-width($two-column-min-device-width) {
      &:nth-child(2n + 2) {
        margin-top: $link-default-margin;
        margin-left: $link-default-margin;

        &:not(:last-child) {
          margin-bottom: $link-default-margin;
        }
      }

      &:nth-child(3n + 3) {
        margin-top: $link-short-row-vertical-margin;
        margin-left: $link-short-row-horizontal-margin;

        &:not(:last-child) {
          margin-bottom: $link-short-row-vertical-margin;
        }
      }
    }

    @include media-min-width($three-column-min-device-width) {
      &:nth-child(3n + 3) {
        margin-top: $link-default-margin;
        margin-left: $link-default-margin;

        &:not(:last-child) {
          margin-bottom: $link-default-margin;
        }
      }

      &:nth-child(5n + 4),
      &:nth-child(5n + 5) {
        margin-top: $link-short-row-vertical-margin;

        &:not(:last-child) {
          margin-bottom: $link-short-row-vertical-margin;
        }
      }

      &:nth-child(5n + 4) {
        margin-left: $link-short-row-horizontal-margin;
      }
    }
  }
}

.IndexPage__link__image {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-size: cover;
  background-position: center;
  opacity: 0.2;
  // FIXME: Disabling transition because it causes the image to break out of
  //        the clip-path during transition on Chrome.
  // transition: opacity $hover-transition-duration;
}

.IndexPage__link__content {
  display: flex;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  flex-direction: column;
  justify-content: center;
  text-align: center;
}

.IndexPage__link__text {
  font-family: Dosis, sans-serif;
  font-weight: 400;
  font-size: 28px;
  text-shadow: $darkest-grey 0 0 6px;
}
