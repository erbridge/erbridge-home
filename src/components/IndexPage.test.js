import { shallow } from 'enzyme';
import React from 'react';

import IndexPage from './IndexPage';

it('renders correctly with all props', () => {
  expect(
    shallow(
      <IndexPage
        title="Test"
        routes={[
          { path: '/a', title: 'A' },
          { link: 'Link to B', path: '/b', title: 'B' },
          {
            image: '/path/to/c.jpg',
            link: 'Link to C',
            path: '/c',
            title: 'C',
          },
          {
            image: '/path/to/d.jpg',
            link: 'Link to D',
            name: 'd_name',
            path: '/d',
            title: 'D',
          },
        ]}
      />,
    ),
  ).toMatchSnapshot();
});

it('renders correctly without optional props', () => {
  expect(
    shallow(
      <IndexPage
        routes={[
          { path: '/a', title: 'A' },
          { link: 'Link to B', path: '/b', title: 'B' },
          {
            image: '/path/to/c.jpg',
            link: 'Link to C',
            path: '/c',
            title: 'C',
          },
          {
            image: '/path/to/d.jpg',
            link: 'Link to D',
            name: 'd_name',
            path: '/d',
            title: 'D',
          },
        ]}
      />,
    ),
  ).toMatchSnapshot();
});

it('renders correctly with empty routes', () => {
  expect(shallow(<IndexPage routes={[]} />)).toMatchSnapshot();
});
