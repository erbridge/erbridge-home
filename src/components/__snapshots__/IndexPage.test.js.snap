// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="IndexPage"
>
  <div
    className="IndexPage__content"
  />
</div>
`;
